position_in_partition.cc

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <seastar/core/print.hh>
+
+#include "position_in_partition.hh"
+#include "utils/exceptions.hh"
+#include "log.hh"
+
+logging::logger plogger("position_in_partition");
+
+const sstring position_in_partition_tracker::current_position_string() {
+    if (is_beginning_of_stream()) {
+        return "beginning of stream";
+    }
```

Comment:
A switch would be nicer.

Reply:
You mean a switch on the `partition_region`?

Reply:
Yes.

Reply:
change in v2 to use operator<